cvat/apps/quality_control/quality_reports.py

```diff
@@ -2082,19 +2108,52 @@ def generate_report(self) -> ComparisonReport:
             ),
         )
         mean_ious = []
+        empty_frame_count = 0
         confusion_matrix_labels, confusion_matrix, _ = self._make_zero_confusion_matrix()
 
-        for frame_id, frame_result in self._frame_results.items():
-            intersection_frames.append(frame_id)
-            conflicts += frame_result.conflicts
+        for frame_result in frame_summaries.values():
             confusion_matrix += frame_result.annotations.confusion_matrix.rows
 
+            if not np.any(frame_result.annotations.confusion_matrix.rows):
+                empty_frame_count += 1
+
             if annotation_components is None:
                 annotation_components = deepcopy(frame_result.annotation_components)
             else:
                 annotation_components.accumulate(frame_result.annotation_components)
+
             mean_ious.append(frame_result.annotation_components.shape.mean_iou)
 
+        annotation_summary = self._compute_annotation_summary(
+            confusion_matrix, confusion_matrix_labels
+        )
+
+        if self.settings.match_empty_frames and empty_frame_count:
+            # Add virtual annotations for empty frames,
+            # they are not included in the confusion matrix
+            annotation_summary.valid_count += empty_frame_count
+            annotation_summary.total_count += empty_frame_count
+            annotation_summary.ds_count += empty_frame_count
+            annotation_summary.gt_count += empty_frame_count
+
+        # Cannot be computed in accumulate()
+        annotation_components.shape.mean_iou = np.mean(mean_ious)
```

Comment:
This doesn't seem right. A mean of means is not necessarily the same as the mean of all underlying values.

(It also seems unrelated to the topic of the PR.)

Reply:
It's how it worked before this PR, I didn't plan to change anything here. Also, we don't have too many options here on how to compute the value.

Reply:
Ah, okay, I didn't see it was moved from a different section.

> Also, we don't have too many options here on how to compute the value.

FWIW, I would imagine you could change the `mean_iou` member of `ComparisonReportAnnotationShapeSummary` to `sum_iou`, and compute the mean on-demand instead.